website/docs/features/lightgbm/about.md

```diff
@@ -122,16 +122,57 @@ and can be tuned with [SparkML's cross
+doesn't affect training, just how data is put into Datasets. It can, however, affect memory usage and overall
+fit/transform time. 
+
+#### Bulk Execution mode
+"Bulk" mode is older and requires accumulating all data in executor memory before creating Datasets. This original mode can cause
+OOM errors for large data, especially since the data must be accumulated in its original double-format size.
+For now, "bulk" mode is the default since "streaming" is new, but SynapseML will eventually make streaming the default once
+it has seen more usage.
+
+#### Streaming Execution Mode
+SynapseML worked with the native LightGBM team to create more efficient LightGBM APIs that work more like common
+streaming APIs and thus don't require loading extra copies of the data into memory. Data is passed directly
+from partitions to Datasets in small "micro-batches", similar to Spark streaming. You can adjust the micro-batch size.
+Smaller sizes reduce memory overhead, but larger sizes avoid as much marshaling back and forth across the native boundary. The default
+100, which still uses far less memory than bulk mode since only 100 rows of data will be loaded at a time. If you have
+a small number of columns, you could increase the batch size, and conversely if
+you have a large number of columns you could decrease it.
+
+Also, the new streaming APIs developed for SynapseML are thread-safe, and allow all partitions in the same executor
+to push data into a shared Dataset in parallel. Because of this, streaming mode always uses the more efficient
+"useSingleDatasetMode=true", creating only one Dataset per executor.
+
+You can explicitly specify Execution Mode and MicroBatch size as parameters.
+
+    val lgbm = new LightGBMClassifier()
+        .setExecutionMode("streaming")
+        .setMicroBatchSize(100)
+        .setLabelCol(labelColumn)
+        .setObjective(binaryObjective)
+        .setUseBarrierExecutionMode(true)
+    ...
+    <train classifier>
+
 ### Barrier Execution Mode
 
 By default LightGBM uses regular spark paradigm for launching tasks and communicates with the driver to coordinate task execution.
 The driver thread aggregates all task host:port information and then communicates the full list back to the workers in order for NetworkInit to be called.
-This requires the driver to know how many tasks there are, and if the expected number of tasks is different from actual this will cause the initialization to deadlock.
-There is a new UseBarrierExecutionMode flag, which when activated uses the barrier() stage to block all tasks.
+This procedure requires the driver to know how many tasks there are, and if the expected number of tasks is different from actual it will cause the initialization to deadlock.
+There's a UseBarrierExecutionMode flag, which when activated uses the barrier() stage to block all tasks.
 The barrier execution mode simplifies the logic to aggregate host:port information across all tasks.
```

Comment:
```suggestion
To avoid this issue use the `UseBarrierExecutionMode` flag, to use Apache Spark's `barrier()` stage to ensure all tasks execute at the same time.
Barrier execution mode simplifies the logic to aggregate `host:port` information across all tasks.
```

Reply:
ok, now you are editing parts I didn't change  :)